### DISCUSSION TITLE
Programmatically Schedule Jobs

### DISCUSSION THREAD

**Discussion by mnouwens** [quarkusio/quarkus#24598] [Q&A]
Hi,

How can I Programmatically schedule job in quarkus with the simple in memory scheduler.

In Thorntail we used: TimerService and Timer

Any equivalent for quarkus?

Thanks

[a comment by an automated account is not shown]

**Comment by mkouba** [comment]
> How can I Programmatically schedule job in quarkus with the simple in memory scheduler.

You can't at the moment. At least not by means of the `quarkus-scheduler` API. You can try the [quarkus-quartz](https://quarkus.io/guides/scheduler-reference#programmatic-scheduling) extension or leverage the Vertx API, i.e. inject [the Vertx instance](https://quarkus.io/guides/vertx#accessing-the-managed-vert-x-instance) and use the [Periodic Timers](https://vertx.io/docs/vertx-core/java/#_periodic_timers) and [One-shot Timers](https://vertx.io/docs/vertx-core/java/#_one_shot_timers). Note that the vertx timers are executed on the event loop so the scheduled code should either not block or [run in a special way](https://vertx.io/docs/vertx-core/java/#blocking_code).
